### PR TITLE
u and v added along with corresponding geometry

### LINKED ISSUE #283: C-grid geometry
## Description
Add C-grid info to the current Geometry relevant to the staggered velocities.

### LINKED ISSUE #284: Add staggered u,v to State and increment
## Description
Add velocities to State/Incr and account for rotated u,v and grid staggering.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
Addition of u,v in field and model and associated geometry (lon, lat and mask).
Slight change of answer in the forecast test when u and v are used and passed between soca and mom6 (type change is not reversible).

### Issue(s) addressed
This is 1 of many pr's that are needed to address epic #282.
closes #283
closes #284  

## Testing
Tested with singularity/gcc and hera/intel18
Changes are covered in the forecast_mom6 test

### CODE REVIEW

File: src/soca/Fields/soca_fields_mod.F90 [hunk 1]
```diff
@@ -45,6 +45,7 @@ module soca_fields_mod
   integer                           :: nz         !< the number of levels
   logical                           :: masked     !< if true, land mask should be applied
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90]: delete `masked`, we don't need it now that we have `mask`... redundant information. 
Just check for `associated(mask)` anywhere `masked` is used.
Also, need to add `self%mask => rhs%mask` inside `soca_field_copy`

File: src/soca/Fields/soca_fields_mod.F90 [hunk 2]
```diff
@@ -45,6 +45,7 @@ module soca_fields_mod
...
   real(kind=kind_real), allocatable :: val(:,:,:) !< the actual data
+  real(kind=kind_real),     pointer :: mask(:,:)  !< field mask
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90]:
```suggestion
  real(kind=kind_real),     pointer :: mask(:,:) => null() !< field mask
```

File: src/soca/Fields/soca_fields_mod.F90 [hunk 3]
```diff
@@ -954,12 +975,13 @@ subroutine soca_fields_read(fld, f_conf, vdate)
...
             select case(field%name)
             ! TODO remove hardcoded variable names here
+            ! TODO Add u and v. Remapping u and v will require interpolating h 
             case ('tocn','socn')
-              if (fld%geom%mask2d(i,j).eq.1) then
-                tsocn_ij = field%val(i,j,:)
-                call remapping_core_h(remapCS, nz, h_common_ij, tsocn_ij,&
-                      &nz, hocn_ij, tsocn2_ij)
-                field%val(i,j,:) = tsocn2_ij
+              if (field%mask(i,j).eq.1) then
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90]:
```suggestion
              if (associated(field%mask) .and. field%mask(i,j).eq.1) then
```

**Review by travissluka — CHANGES_REQUESTED** [review]

File: src/soca/Fields/soca_fields_mod.F90 (hunk 1 above)
Comment (guillaumevernieres) [reply src/soca/Fields/soca_fields_mod.F90]: Good catch, thanks. Re-tested on hera with suggested changes.

**Review by travissluka — APPROVED** [review]

**Pull request merged**